tapfreighter/interface.go

```diff
@@ -252,6 +252,12 @@ type TransferOutput struct {
 // ShouldDeliverProof returns true if a proof corresponding to the subject
 // transfer output should be delivered to a peer.
 func (out *TransferOutput) ShouldDeliverProof() (bool, error) {
+	// If the proof delivery is already complete, we don't need to deliver
+	// a proof.
+	if out.ProofDeliveryComplete == fn.Some(true) {
```

Comment:
`if out.ProofDeliveryComplete..UnwrapOr(false) {` is more complicated in my mind. I'm not sure if it's correct? Shouldn't it be:
```
if out.ProofDeliveryComplete.UnwrapOr(true) {
```
Because if `out.ProofDeliveryComplete` is `None` then we want the conditional to evaluate to true (so that `ShouldDeliverProof` returns false).

Reply:
Yeah, depending on whether the desired outcome for `None` is true or false that should be used.
I agree that it isn't necessarily more readable. I'm just not sure if comparing values of `fn.Option` always returns the expected result. For instance if the value is not a comparable value, it would compare pointers instead of the content. So even though we use primitive/comparable values here, I think it's bad practice to do that.

Reply:
I was wrong, `if out.ProofDeliveryComplete..UnwrapOr(false) {` is correct because we're using `ShouldDeliverProof` to set the value of `ProofDeliveryComplete` and we only return false if `ProofDeliveryComplete` shows that the proof has been delivered. And not if `ProofDeliveryComplete` is none/unset.